### ISSUE TITLE
Example 4: Create a team from group EducationClass

### ISSUE THREAD

**Issue by ashleywb** [microsoftgraph/microsoft-graph-docs#5537]
I've tried Example 4: Create a team from group but get the following error : 
Team with template 'educationClass' can not be provisioned using a group which does not belong to a class.
I've tried creating the group via https://docs.microsoft.com/en-us/graph/api/group-post-groups?view=graph-rest-beta&tabs=http and then running example 4 and get this error.

I've also tried creating the group via this method https://docs.microsoft.com/en-us/graph/api/educationroot-post-classes?view=graph-rest-beta&tabs=http and still get the same error

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 6f249664-3b6d-e3f9-9017-1aa0121fe949
* Version Independent ID: c7aeea07-12b6-45c9-8fb9-f942974811fa
* Content: [Create a team - Microsoft Graph beta](https://docs.microsoft.com/en-us/graph/api/team-post?view=graph-rest-beta#example-4-create-a-team-from-group)
* Content Source: [api-reference/beta/api/team-post.md](https://github.com/microsoftgraph/microsoft-graph-docs/blob/master/api-reference/beta/api/team-post.md)
* Product: **microsoft-teams**
* Technology: **microsoft-graph**
* GitHub Login: @nkramer
* Microsoft Alias: **MSGraphDocsVteam**

**Comment by fweier** [comment]
I would like to confirm this: Creating an educationClass first via the "Create educationClass"  POST call to /education/classes still does not let me POST a Team to that group later. The error still says "Team with template \'educationClass\' can not be provisioned using a group which does not belong to a class."

Is this intentional ?

**Comment by ehussey** [comment]
It doesn't seem to work with binding to the standard groups API or the endpoint for retrieving the linked group from an educationClass.  I got it working by creating the educationClass, using /education/classes/{id}/group to retrieve the group object, then feeding the group object itself into the /teams POST request.  For example, here's what I'm sending in that POST request:
{
    "displayName": <displayName>,
    "template@odata.bind": "https://graph.microsoft.com/beta/teamsTemplates('<template>')",
    "owners@odata.bind": ["https://graph.microsoft.com/beta/users/<owner id>"],
    "group": <group object>
}

**Comment by jeffangama** [comment]
@ehussey  can you share full code of it ?
When i do put on 

`https://graph.microsoft.com/v1.0/education/classes/xxxxx-xxxxx-xxxxxx-xxxxxx-xxxxxx-xxxxx/group/team`

body :
```
{  
  "memberSettings": {
    "allowCreatePrivateChannels": true,
    "allowCreateUpdateChannels": true
  },
  "messagingSettings": {
    "allowUserEditMessages": true,
    "allowUserDeleteMessages": true
  },
  "funSettings": {
    "allowGiphy": true,
    "giphyContentRating": "strict"
  }
}
```

I get 

`{
    "error": {
        "code": "BadRequest",
        "message": "Write requests are only supported on contained entities",
        "innerError": {
            "date": "2020-08-14T13:17:53",
            "request-id": "41102cfa-9f63-468c-b826-30ad1922f4dc"
        }
    }
}`

**Comment by ehussey** [comment]
You have to use the beta API, if I recall correctly.
After more experimentation, I found that I could create classroom teams more reliably if I skipped creating an educationClass first, left out most of the initial parameters, used my desired mailNickname as the initial displayName, and then changed the displayName to something user-friendly afterwards.  I send a `POST` to `https://graph.microsoft.com/beta/teams` to create the team:
```
{
  "displayName": "Teams_Classroom_12345_34FA8F",
  "template@odata.bind": "https://graph.microsoft.com/beta/teamsTemplates('educationClass')",
  "owners@odata.bind": ["https://graph.microsoft.com/beta/users/<SERVICE_ACCOUNT_GUID>"]
}
```
(I set the initial owner to a service account, so the end user doesn't get confused by the initial display name)
The call returns a "location" header for the creation status, so I check that URL every 5-10 seconds until I get a "succeeded" response.  Sometimes this can take a long time, so I stop trying after a minute or so and set up a process to check the URL once every hour.  The "succeeded" response includes a targetResourceId GUID that points to the newly-created team, so I use that to send a `PATCH` request to `https://graph.microsoft.com/beta/groups/<TARGET_RESOURCE_ID>` to change the displayName:
```
{
    "displayName": "Ellie's test Team"
}
```
A `POST` to `https://graph.microsoft.com/beta/groups/$teamId/owners/$ref` to add the actual owner for the team:
```
{
    "@odata.id" => "https://graph.microsoft.com/beta/users/<OWNER_ACCOUNT_GUID>"
}
```
and then a `DELETE` to `https://graph.microsoft.com/beta/groups/$teamId/owners/<SERVICE_ACCOUNT_GUID>/$ref` to remove the service account.

**Comment by jeffangama** [comment]
Thank you for your quick answer, and then the group would be created correctly ?
My goal is to do like classic teams : 
1. Create the group
2. Add a team to the group

To workaround a feature where we need to have the TEAM Calendar (group), showing in outlook : https://techcommunity.microsoft.com/t5/microsoft-teams-ama/why-the-teams-calendar-not-visible-in-outlook/td-p/300155
In this way I avoid using powershell to enable the calendar visibility in Outlook

**Comment by jeffangama** [comment]
**Edit:** 
I got it working your way,

` creating the educationClass, using /education/classes/{id}/group to retrieve the group object, then feeding the group object itself into the /teams POST request. For example, here's what I'm sending in that POST request:
{
"displayName": DISPLAYNAME,
"template@odata.bind": "https://graph.microsoft.com/beta/teamsTemplates('TEMPLATE')",
"owners@odata.bind": ["https://graph.microsoft.com/beta/users/OWNER_ID"],
"group": GROUP_OBJECT
}`

 but setting the userID instead of Email
` "owners@odata.bind": ["https://graph.microsoft.com/beta/users/admin@mytenant.onmicrosoft.com"]`

 but i want to do this from PowerAutomate, a bit of a complex approach for now...

**Comment by chakudes** [comment]
Hello,
I can't create a private channel in application context even if I try to add an owner at the creation. Can someone help please.

Error  message:  Cannot create private channel, no members specified when calling from an Application

**Comment by Prithvi-MSFT** [comment]
Hi @ashleywb ,
Are you still facing this issue ? We tried it at our end and it worked for us. First we created new group using below example
https://docs.microsoft.com/en-us/graph/api/group-post-groups?view=graph-rest-beta&tabs=http#example-1-create-a-microsoft-365-group

Then we used the created group to create a new team as mentioned in example 4 of below document.
https://docs.microsoft.com/en-us/graph/api/team-post?view=graph-rest-beta&tabs=http#example-4-create-a-team-from-group

Let us know if you are still facing any issue.

Thanks
